core/runtime/src/main/java/io/quarkiverse/langchain4j/RegisterAiService.java

```diff
@@ -0,0 +1,156 @@
+package io.quarkiverse.langchain4j;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.util.function.Supplier;
+
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.memory.ChatMemory;
+import dev.langchain4j.memory.chat.ChatMemoryProvider;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.moderation.ModerationModel;
+import dev.langchain4j.retriever.Retriever;
+import dev.langchain4j.service.AiServices;
+
+/**
+ * Used to create Langchain4j's {@link AiServices} in a declarative manner that the application can then use simply by
+ * using the class as a bean.
+ * Under the hood Langchain4j's {@link AiServices#builder(Class)} is called
+ * while also providing the builder with the proper {@link ChatLanguageModel}, {@code tools} beans,
+ * {@link ChatMemory} or {@link ChatMemoryProvider}, {@link ModerationModel} and {@link Retriever}
```

Comment:
An idea for later - we could invoke a customizer bean if available which would receive the builder and return the builder. The advantage is to allow programmatic customization.

Reply:
Definitely